src/Bundle/Resources/config/services/expression_language.xml

```diff
@@ -30,6 +30,11 @@
             <tag name="sylius.routing_variables" />
         </service>
 
+        <service id="sylius.expression_language.variables.sylius_repositories" class="Sylius\Resource\Symfony\ExpressionLanguage\SyliusRepositoriesVariables">
+            <argument type="tagged_locator" tag="sylius.repository" />
+            <tag name="sylius.resource_factory_variables" />
```

Comment:
that tag allow you to use this variables on your factory arguments (handle by the ArgumentParser)